core/common/src/main/java/alluxio/underfs/Fingerprint.java

```diff
@@ -73,6 +80,36 @@ public static Fingerprint create(String ufsName, UfsStatus status) {
     if (status == null) {
       return new Fingerprint(Collections.emptyMap());
     }
+    return new Fingerprint(Fingerprint.createTags(ufsName, status));
+  }
+
+  /**
+   * Parses the input string and returns the fingerprint object.
+   *
+   * @param ufsName the name of the ufs, should be {@link UnderFileSystem#getUnderFSType()}
+   * @param status the {@link UfsStatus} to create the fingerprint from
+   * @param acl the {@link AccessControlList} to create the fingerprint from
+   * @return the fingerprint object
+   */
+  public static Fingerprint create(String ufsName, UfsStatus status, AccessControlList acl) {
+    if (status == null) {
+      return new Fingerprint(Collections.emptyMap());
+    }
+    Map<Tag, String> tagMap = Fingerprint.createTags(ufsName, status);
+    if (acl != null) {
+      tagMap.put(Tag.ACL, acl.toString());
```

Comment:
I forgot, but is `AccessControlList.toString()` a deterministic string?

Reply:
Good catch!  I changed the map to use `TreeMap` rather than `HashMap` to be deterministic. I think the slower insertion speed is ok since I can't imagine a single `AccessControlList` having too many ACL entries.